### PR TITLE
Adding transportation in trip services

### PR DESCRIPTION
**Pull request by RobinCsl**
Add transportation menu item

Add current location item

Remove wrong icons

Add deeplinking

### CODE REVIEW

File: app/MMB/src/scenes/tripServices/transportation/AddressLocationInput.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
+// @flow
+
+import * as React from 'react';
+import { TextInput as OriginalTextInput, View } from 'react-native';
+import { Text, StyleSheet, TextIcon } from '@kiwicom/mobile-shared';
+import { Translation } from '@kiwicom/mobile-localization';
+import { defaultTokens } from '@kiwicom/mobile-orbit';
+
+type Props = {|
+  +placeholder?: React.Element<typeof Translation>,
```
Comment (tbergq) [inline app/MMB/src/scenes/tripServices/transportation/AddressLocationInput.js]: We export TranslationType from localization

File: app/MMB/src/scenes/tripServices/transportation/CurrentPositionButton.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    width: 40,
+    height: 40,
+    borderColor: defaultTokens.paletteWhite,
+    borderRadius: 40,
+    justifyContent: 'center',
+    alignItems: 'center',
+    backgroundColor: defaultTokens.paletteWhite,
+    shadowOpacity: 0.3,
+    shadowRadius: 2,
+    shadowColor: Color.black,
```
Comment (tbergq) [inline app/MMB/src/scenes/tripServices/transportation/CurrentPositionButton.js]: We are not supposed to use this color, you should use `paletteInkDark`

File: app/MMB/src/scenes/tripServices/transportation/TransportationMap.js [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+
+type Region = {|
+  ...Coordinate,
+  +latitudeDelta: number,
+  +longitudeDelta: number,
+|};
+
+type Markers = {
+  +markerA: ?Coordinate,
+  +markerB: ?Coordinate,
```
Comment (tbergq) [inline app/MMB/src/scenes/tripServices/transportation/TransportationMap.js]: could this also be exact type?

File: app/MMB/src/scenes/tripServices/transportation/TransportationMap.js [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+
+  onRegionChange = (region: Region) => {
+    this.setState({ region });
+  };
+
+  buildWhitelabelURL = (whitelabelURL: string) => {
+    const { markerA, markerB } = this.state.markers;
+    if (markerA != null && markerB != null) {
+      const URLbase = whitelabelURL.replace(/\?.*/, '');
+      const URLparameters = whitelabelURL.replace(/.*\?/, '');
```
Comment (tbergq) [inline app/MMB/src/scenes/tripServices/transportation/TransportationMap.js]: maybe we should rather just change how we return the whitelabel url from the graphql?

File: app/MMB/src/scenes/tripServices/transportation/TransportationMap.js [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+        markerB: {
+          latitude,
+          longitude,
+        },
+      },
+    }));
+  };
+
+  getFormattedAddress = async ({ latitude, longitude }: Coordinate) => {
+    const address = await fetch(
```
Comment (tbergq) [inline app/MMB/src/scenes/tripServices/transportation/TransportationMap.js]: please add this to graphql, all calls should go through there

File: app/MMB/src/scenes/tripServices/transportationMenuItem/TransportLocationItem.js [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  +onPress: ({
+    location: { lat: number, lng: number },
+    whitelabelURL: string,
+  }) => void,
+  +whitelabelURL: string,
+|};
+
+class LocationItem extends React.Component<Props> {
+  onPress = () => {
+    const lat = idx(this.props.data, _ => _.location.lat) || 0;
```
Comment (tbergq) [inline app/MMB/src/scenes/tripServices/transportationMenuItem/TransportLocationItem.js]: this is not the best fallback, https://en.wikipedia.org/wiki/Null_Island

File: ios/reactNativeApp.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -773,6 +773,7 @@
 			buildSettings = {
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CLANG_ENABLE_MODULES = YES;
+				CODE_SIGN_IDENTITY = "iPhone Developer";
```
Comment (tbergq) [inline ios/reactNativeApp.xcodeproj/project.pbxproj]: Why this change?

cc @ferrannp?

File: app/MMB/src/navigation/AddressPickerScreen.js [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+  MenuItem,
+  type NavigationType,
+} from '@kiwicom/mobile-navigation';
+import { TextIcon } from '@kiwicom/mobile-shared';
+
+import AddressLocationInput from '../scenes/tripServices/transportation/AddressLocationInput';
+
+type Props = {
+  +navigation: NavigationType,
+  +currentAddress: ?string,
```
[the remark this answers is not in the document]
Comment (RobinCsl) [reply app/MMB/src/navigation/AddressPickerScreen.js]: It should be, thanks!

File: app/MMB/src/navigation/AddressPickerScreen.js [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+    };
+  };
+  render() {
+    const { currentAddress } = this.props;
+    return (
+      <ScrollView>
+        {currentAddress != null && (
+          <MenuItem
+            onPress={todo}
+            actionIcon={<View />}
```
[the remark this answers is not in the document]
Comment (RobinCsl) [reply app/MMB/src/navigation/AddressPickerScreen.js]: I put this view as TODO for now. I will remove that code and stash it somewhere.

File: app/MMB/src/navigation/NavigationStack.js [hunk 10]
```diff
@@ -88,6 +108,9 @@ export default StackNavigator(
       screen: TravelDocumentModalStack,
     },
     // This is not good enough, it is complicated, needs to be improved later
```
[the remark this answers is not in the document]
Comment (RobinCsl) [reply app/MMB/src/navigation/NavigationStack.js]: Done 😃

File: app/MMB/src/scenes/tripServices/transportation/TransportationMap.js (hunk 4 above)
Comment (RobinCsl) [reply app/MMB/src/scenes/tripServices/transportation/TransportationMap.js]: Like you send the arguments, and it gives you back the correct URL?

File: app/MMB/src/scenes/tripServices/transportation/TransportationMap.js (hunk 5 above)
Comment (RobinCsl) [reply app/MMB/src/scenes/tripServices/transportation/TransportationMap.js]: Ok, I will add it there.

File: app/MMB/src/scenes/tripServices/transportationMenuItem/TransportLocationItem.js (hunk 6 above)
Comment (RobinCsl) [reply app/MMB/src/scenes/tripServices/transportationMenuItem/TransportLocationItem.js]: What do you suggest I take then? It really depends on each location, so it's hard to choose one that would make sense? Brno as Kiwi was born in Brno?

File: ios/reactNativeApp.xcodeproj/project.pbxproj (hunk 7 above)
Comment (RobinCsl) [reply ios/reactNativeApp.xcodeproj/project.pbxproj]: This was from Luke. I don't know. I can remove that?

File: app/MMB/src/scenes/tripServices/transportation/TransportationMap.js (hunk 3 above)
Comment (RobinCsl) [reply app/MMB/src/scenes/tripServices/transportation/TransportationMap.js]: Of course!

File: app/MMB/src/scenes/tripServices/transportation/CurrentPositionButton.js (hunk 2 above)
Comment (RobinCsl) [reply app/MMB/src/scenes/tripServices/transportation/CurrentPositionButton.js]: Thanks!

File: app/MMB/src/scenes/tripServices/transportation/AddressLocationInput.js [hunk 11]
```diff
@@ -0,0 +1,99 @@
+// @flow
+
+import * as React from 'react';
+import { TextInput as OriginalTextInput, View } from 'react-native';
```
[the remark this answers is not in the document]
Comment (RobinCsl) [reply app/MMB/src/scenes/tripServices/transportation/AddressLocationInput.js]: I will redo this, it seems difficult to have the input span all the length of the header if we use our component from shared.

File: app/MMB/src/scenes/tripServices/transportation/TransportationMap.js (hunk 4 above)
Comment (tbergq) [reply app/MMB/src/scenes/tripServices/transportation/TransportationMap.js]: I mean just send the basic, here it is being picked apart and put back together. lets just send `https://url.withouth.parameters/`

File: app/MMB/src/scenes/tripServices/transportationMenuItem/TransportLocationItem.js (hunk 6 above)
Comment (tbergq) [reply app/MMB/src/scenes/tripServices/transportationMenuItem/TransportLocationItem.js]: Maybe show an alert say there is something wrong 🤷‍♂️  better than sending coordinates in the middle of the ocean, don't you think?

**Comment by tbergq** [comment]
Also could you rewrite the commit message, reading the commit history, this message will not tell me what has changed.

File: app/MMB/src/scenes/tripServices/transportation/TransportationMap.js (hunk 5 above)
Comment (RobinCsl) [reply app/MMB/src/scenes/tripServices/transportation/TransportationMap.js]: Decided against it, for now.

File: packages/shared/src/DropMarker.js [hunk 12]
```diff
@@ -24,13 +25,13 @@ type Props = {|
...
  */
-export default function DropMarker({ size = 50 }: Props) {
+export default function DropMarker({ size = 50, color }: Props) {
   const styles = createStyles(size);
   return (
     <Icon
       name="place"
       size={size}
-      color={defaultTokens.paletteProductNormal}
+      color={color != null ? color : defaultTokens.paletteProductNormal}
```
Comment (tbergq) [inline packages/shared/src/DropMarker.js]: how about using defaultProps for color?
